File: server_integration_tests/java/java_server/src/main/java/dev/graphlink/javaserver/impl/DeleteArticleServiceImpl.java
package dev.graphlink.javaserver.impl;

import dev.graphlink.javaserver.generated.services.DeleteArticleService;
import org.springframework.stereotype.Service;

@Service
public class DeleteArticleServiceImpl implements DeleteArticleService {

    @Override
    public Boolean deleteArticle(String id) {
        return Data.articles.removeIf(a -> a.getId().equals(id));
    }
}

File: server_integration_tests/servers/kotlin_server_suspend/src/main/kotlin/dev/graphlink/kotlinserversuspend/impl/DeleteArticleServiceImpl.kt
package dev.graphlink.kotlinserversuspend.impl

import dev.graphlink.kotlinserversuspend.generated.services.DeleteArticleService
import org.springframework.stereotype.Service

@Service
class DeleteArticleServiceImpl : DeleteArticleService {
    override suspend fun deleteArticle(id: String): Boolean = Data.articles.removeAll { it.id == id }
}
